bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/XMLItem.java

```diff
@@ -464,7 +465,14 @@ public void serialize(OutputStream outputStream) {
             if (outputStream instanceof BallerinaXMLSerializer) {
                 ((BallerinaXMLSerializer) outputStream).write(this);
             } else {
-                (new BallerinaXMLSerializer(outputStream)).write(this);
+                BallerinaXMLSerializer ballerinaXMLSerializer = new BallerinaXMLSerializer(outputStream);
+                ballerinaXMLSerializer.write(this);
+                try {
+                    ballerinaXMLSerializer.flush();
+                    ballerinaXMLSerializer.close();
+                } catch (IOException e) {
+                    throw new BallerinaException(e);
```

Comment:
Shouldn't we use handleXmlException here also?

Reply:
Yes, we should let the outer catch block to catch this IOException.